inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx

```diff
@@ -1,28 +1,406 @@
 
+    private _anchorPoints: { point: Vector2, anchor: Vector2 }[] = [];
+    private _keyframes: Vector2[] = [];
     constructor(props: IAnimationCurveEditorComponentProps) {
         super(props);
+        this.state = { selected: this.props.animations[0], isOpen: true, currentPathData: this.getPathData(this.props.animations[0]), anchorPoints: this._anchorPoints, keyframes: this._keyframes }
+    }
+
+    getAnimationProperties(animation: Animation) {
+
+        let easingFunction: EasingFunction = animation.getEasingFunction() as EasingFunction;
+        let easingType = easingFunction.constructor.name;
+        let easingMode = easingFunction.getEasingMode();
+
+        return { easingType, easingMode }
+
+    }
+
+    getPathData(animation: Animation) {
+
+        const { easingMode, easingType } = this.getAnimationProperties(animation);
+
+        var easingFunction = animation.getEasingFunction();
+        console.log(easingFunction);
+
+        const keyframes = animation.getKeys();
+        if (keyframes === undefined) {
+            return "";
+        }
+
+        const startKey = keyframes[0];
+
+        // This will change when Graph width becomes dinamic
+        const heightScale = 100;
+
+        // This assumes the startkey is always 0... we will change this
+        let middle = (startKey.value / 2) * heightScale;
+
+        // START OF LINE/CURVE
+        let data: string | undefined = `M0, ${middle}`;
+
+        if (easingType === undefined){
+            data = this.linearInterpolation(keyframes, data, heightScale, middle);
+        } else {
+            var bezier = this.getEasingBezierValues(easingMode, easingType, easingFunction);
+            data = bezier && this.bezierEasePath(keyframes, data, heightScale, middle, [bezier.x, bezier.y], [bezier.z, bezier.w], easingMode);
+        }
+
+        return data;
+
+    }
+
+    linearInterpolation(keyframes: IAnimationKey[], data: string, heightScale: number, middle: number): string {
+        keyframes.forEach((key, i) => {
+            if (i !== 0) {
+                data += ` L${key.frame} ${heightScale - (key.value * middle)}`
+            }
+
+        });
+        return data;
+    }
+
+    getEasingBezierValues(easingMode: number, easingType: string, easingFunction: IEasingFunction): Vector4 | undefined {
+
+        // X, Y, W, Z
+        let easingFunctionMode = `${easingType}_${easingMode}`;
+        let bezierValues:Vector4 | undefined;
+
+        if (easingType === 'BezierCurveEase'){
+            let easeF = easingFunction as BABYLON.BezierCurveEase;   
+            bezierValues = new Vector4(easeF.x1, easeF.y1, easeF.x2, easeF.y2);
+        } else {
+               
+        switch (easingFunctionMode){
+            case 'CircleEase_0':
+                bezierValues = new Vector4(0.55,0,1,0.45);//cubic-bezier(0.55, 0, 1, 0.45);

```

Comment:
I'm afraid this does not scale as users can provide their own easing functions. So I feel like we will need to directly use the easinCore function to draw the curves

Reply:
Yes, this switch only works for the predefined functions. The idea is to have another way to support custom easing functions. This is especially true for Bounce and Elastic.

Reply:
Do you see any value of having the predefined functions here? The context for these here, is that if someone uses a predefined function and comes to the editor they can see their animation and then tweak it, thus creating a custom curve animation in the editor. But yes as you mention above the idea would be to generate the custom curves with easinCore.

Reply:
I feel like having the predefined function here seems low and maintenance heavy so I would recommend to only use easeInCore

Reply:
cool, roger 😎

Reply:
Sorry, I've been working on creating and editing new animations with the curve editor. I've been working on a separate thread for this. I don't know if I should just cancel this PR.

Reply:
So, I'm a little bit lost here. For this PR, do you think it would be better to just add a way to view the "custom" animations other people create?

Reply:
Well we can cancel it maybe. I would like to avoid having predefined function written in the code

Reply:
So, after thinking about it more, the main trouble I have here is figuring out how to draw the SVG path with only the "easeInCore" function. Any suggestions? For the other curves I take the bezier curve values and based on that I can draw the path. So the idea would be to have a way to get the control points from the easingFunction.

Reply:
we can probably sample the curve between two control points by evaluating the easingCore along the way?

Reply:
I can try that yes. I'll focus on that and clean this PR, so we don't have predefined functions here.

Reply:
I'm going to try this https://math.stackexchange.com/questions/2459135/find-control-points-of-b%C3%A9zier-curve-in-order-to-approximate-a-function

Reply:
I think I finally got it to work. I'll add a new commit early tomorrow. I follow this guide http://polymathprogrammer.com/2007/06/27/reverse-engineering-bezier-curves/